api/README.md

```diff
@@ -1 +1,269 @@
+- `204 No Content` - The request didn't result in a creation or multiple resources had been created.
+
+#### PUT
+- `200 OK` - An existing resource has been modified which is retrieved.
+- `201 Created` - A new resource has been created which is retrieved.
+- `204 No Content` - An existing resource has been modified.
+
+#### PATCH
+- `200 OK` - An existing resource has been modified which is retrieved.
+- `204 No Content` - An existing resource has been modified.
+- `404 Not Found` - The requested resource does not exist (in the current context).
+
+#### DELETE
+- `200 OK` - An existing resource has been deleted which is retrieved.
+- `202 Accepted` - An existing resource has been marked for deletion.
+- `204 No Content` - An existing resource has been deleted.
+
+#### In all cases
+- `400 Bad Request` - The request is malformed, such as if the body does not parse or it has validation errors.
+- `401 Unauthorized` - When no or invalid authentication details are provided. Also useful to trigger an auth popup on web or take the user to the login screen on mobile.
+- `403 Forbidden` - When authentication succeeded but authenticated user doesn't have access to the resource.
+- `405 Method Not Allowed` - When an HTTP method is being requested that isn't allowed for the resource or the authenticated user.
+- `500 Internal Server Error` The server encountered an unexpected condition, which prevented it from fulfilling the request. 
+- `501 Not Implemented` - The server does not support the functionality required to fulfill the request. This is the appropriate response when the server does not recognize the request method and is not capable of supporting it for any resource.
+- `503 Service Unavailable` - The server is currently unable to handle the request due to a temporary overloading or maintenance of the server. The implication is that this is a temporary condition which will be alleviated after some delay. If known, the length of the delay may be indicated in a Retry-After header. If no Retry-After is given, the client **should** handle the response as it would for a 500 response.
+
+### Allow overriding HTTP method
+
+Some proxies support only POST and GET methods. To support a RESTful API with these limitations, the API needs a way to override the HTTP method.
+Use the custom HTTP Header X-HTTP-Method-Override to override the POST Method.
+
+### Error handling
+
+The API should provide a useful error message in a known consumable format. The representation of an error should be no different than the representation of any resource, just with its own set of fields.
+The API should always return sensible HTTP status codes. A JSON error body should provide a few things for the developer - a useful error message, a unique error code (that can be looked up for more details in the docs) and possibly a detailed description. JSON output representation for something like this would look like:
+
+```javascript
+{
+  "code" : 1234,
```

Comment:
We saw cases where the "code" tag was used for different purposes, other than errors. In order to make it clearer and easier to differentiate valid responses with error content and normal content, it would be a bit more straight forward if you guys would use "error" or "error_code".